src/pages/TransactionDetails/components/TransactionInfo/TransactionDetailsPanel/TransactionDetailsPanel.tsx

```diff
@@ -0,0 +1,352 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import BigNumber from 'bignumber.js';
+import { useSelector } from 'react-redux';
+
+import {
+  FormatAmount,
+  ShardSpan,
+  NetworkLink,
+  TimeAgo,
+  TransactionStatus,
+  DetailItem,
+  Trim,
+  CopyButton,
+  TransactionAction,
+  FormatUSD,
+  AccountLink,
+  ShardLink,
+  TransactionIcons
+} from 'components';
+import {
+  addressIsBech32,
+  formatAmount,
+  formatDate,
+  isContract,
+  getTransactionMethod,
+  getTotalTxTokenUsdValue,
+  getDisplayReceiver,
+  getTransactionVisibleOperations
+} from 'helpers';
+import { faClock, faSpinner } from 'icons/regular';
+import {
+  OperationsList,
+  ScResultsList,
+  DataField,
+  TransactionErrorDisplay,
+  TransactionWarningMessage
+} from 'pages/TransactionDetails/components';
+import { activeNetworkSelector } from 'redux/selectors';
+import {
+  TransactionType,
+  TransactionActionCategoryEnum,
+  TransactionApiStatusEnum
+} from 'types';
+
+export const getFee = (transaction: TransactionType) => {
+  const bNgasPrice = new BigNumber(transaction.gasPrice);
+  const bNgasUsed = new BigNumber(transaction.gasUsed);
+  const output = bNgasPrice.times(bNgasUsed).toString();
+
+  return output;
+};
+
+export const TransactionDetailsPanel = ({
+  transaction
+}: {
+  transaction: TransactionType;
+}) => {
+  const { egldLabel } = useSelector(activeNetworkSelector);
+  const { receiver, receiverAssets } = getDisplayReceiver(transaction);
+
+  const isTxPending =
+    (transaction?.status &&
+      transaction.status.toLowerCase() === TransactionApiStatusEnum.pending) ||
+    transaction.pendingResults;
+
+  const transactionFee =
+    transaction.fee === undefined && transaction.gasUsed === undefined
+      ? 'N/A'
+      : formatAmount({
+          input: transaction.fee ? transaction.fee : getFee(transaction),
+          showLastNonZeroDecimal: true
+        });
+
+  const txValue = formatAmount({
+    input: transaction.value,
+    showLastNonZeroDecimal: true
+  });
+
+  const visibleOperations = getTransactionVisibleOperations(transaction);
+  const totalTxTokenUsdValue = getTotalTxTokenUsdValue(transaction);
+  const showTotalTxTokenUsdValue =
+    totalTxTokenUsdValue !== new BigNumber(0).toString();
+
+  return (
+    <>
+      <DetailItem title='Hash'>
+        <div className='d-flex align-items-center text-break-all text-neutral-100'>
+          <TransactionIcons transaction={transaction} showStatus={false} />
+          {transaction.txHash}
+          <CopyButton text={transaction.txHash} />
+        </div>
+      </DetailItem>
+
+      {transaction?.status && (
+        <DetailItem title='Status'>
+          <div className='d-flex flex-wrap align-items-center'>
+            <TransactionStatus transaction={transaction} />
+          </div>
+        </DetailItem>
+      )}
+
+      <DetailItem title='Age' className='text-neutral-400'>
+        {transaction.timestamp !== undefined ? (
```

Comment:
Could this not be written as `transaction.timestamp &&` so as to avoid other falsey values such as `null`? 👍

Reply:
Awesome. 🔥